File: 238. Product of Array Accept Self/main.cpp
#include <vector>
using namespace std;
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>product;
        int leftProduct = 1;
        for(int x=0;x<nums.size(); x++){
            product.push_back(leftProduct);
            leftProduct *= nums[x];
        }
        int rightProduct=1;
        for(int x =nums.size()-1; x>=0;x--){
            product[x] *= rightProduct;
            rightProduct *= nums[x];
        }
        return product;
    }
};

File: 238. Product of Array Accept Self/main.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length = len(nums)
        answer = []

        # Calculate left products
        left_product = 1
        for i in range(length):
            answer.append(left_product)
            left_product *= nums[i]

        # Calculate right products and the final result
        right_product = 1
        for i in range(length - 1, -1, -1):
            answer[i] *= right_product
            right_product *= nums[i]

        return answer
